typing/typecore.ml

```diff
@@ -224,6 +224,49 @@ let rcp node =
   node
 ;;
 
+module Desugar : sig
```

Comment:
Since this module is not really related to typing and is only transforming `Parsetree`s, I think it would make sense to move it to `parsing`.

Reply:
I've moved this module in parsing/ast_desugar.ml (keeping the same copyright header as parser.mly since the code originates from there)